src/AppInstallerRepositoryCore/Rest/RestClient.cpp

```diff
@@ -80,26 +83,45 @@ namespace AppInstaller::Repository::Rest
         return *commonVersions.rbegin();
     }
 
-    std::unique_ptr<Schema::IRestClient> RestClient::GetSupportedInterface(const std::string& api, const Version& version)
+    std::unique_ptr<Schema::IRestClient> RestClient::GetSupportedInterface(
+        const std::string& api, const std::unordered_map<utility::string_t, utility::string_t>& additionalHeaders, const Version& version)
     {
         if (version == Version_1_0_0)
         {
             return std::make_unique<Schema::V1_0::Interface>(api);
         }
        
+        // TODO: USE additionalHeaders with V1.1 changes.
+        (void)additionalHeaders;
+
         THROW_HR(APPINSTALLER_CLI_ERROR_RESTSOURCE_INVALID_VERSION);
     }
 
-    RestClient RestClient::Create(const std::string& restApi, const HttpClientHelper& helper)
+    RestClient RestClient::Create(const std::string& restApi, std::optional<std::string> customHeader, const HttpClientHelper& helper)
     {
         utility::string_t restEndpoint = RestHelper::GetRestAPIBaseUri(restApi);
         THROW_HR_IF(APPINSTALLER_CLI_ERROR_RESTSOURCE_INVALID_URL, !RestHelper::IsValidUri(restEndpoint));
 
-        IRestClient::Information information = GetInformation(restEndpoint, helper);
+        auto headers = GetHeaders(customHeader);
+
+        IRestClient::Information information = GetInformation(restEndpoint, headers, helper);
         std::optional<Version> latestCommonVersion = GetLatestCommonVersion(information, WingetSupportedContracts);
         THROW_HR_IF(APPINSTALLER_CLI_ERROR_UNSUPPORTED_RESTSOURCE, !latestCommonVersion);
 
-        std::unique_ptr<Schema::IRestClient> supportedInterface = GetSupportedInterface(utility::conversions::to_utf8string(restEndpoint), latestCommonVersion.value());
+        std::unique_ptr<Schema::IRestClient> supportedInterface = GetSupportedInterface(utility::conversions::to_utf8string(restEndpoint), headers, latestCommonVersion.value());
         return RestClient{ std::move(supportedInterface), information.SourceIdentifier };
     }
+
+    std::unordered_map<utility::string_t, utility::string_t> RestClient::GetHeaders(std::optional<std::string> customHeader)

```

Comment:
>GetHeaders [](http://example.com/codeflow?start=73&length=10)

In my rest 1.1 change, I'm trying to reduce exposing small utility functions as static member of RestClient class, can this go to anonymous namespace in this file?